invite/src/invite_bot.ts

```diff
@@ -16,12 +16,24 @@
 
 import {Octokit} from '@octokit/rest';
 
+import {dbConnect} from './db';
 import {GitHub} from './github';
 import {ILogger, Invite, InviteAction} from './types';
 import {InvitationRecord} from './invitation_record';
 
-// TODO: Enable after filling in implementations.
-/* eslint-disable @typescript-eslint/no-unused-vars */
+const INVITE_MACROS: Record<string, InviteAction> = {
+  invite: InviteAction.INVITE,
+  tryassign: InviteAction.INVITE_AND_ASSIGN,
+};
+
+// Regex source: https://github.com/shinnn/github-username-regex
+const USERNAME_PATTERN: string = '[a-z\\d](?:[a-z\\d]|-(?=[a-z\\d])){0,38}';
+const MACRO_PATTERN: string = `/(${Object.keys(INVITE_MACROS).join('|')})`;
+const FULL_MACRO_REGEX: RegExp = new RegExp(
+  // (?<!\\s) ensures the macro is not preceded by a non-space character.
+  `(?<!\\S)${MACRO_PATTERN} @${USERNAME_PATTERN}`,
```

Comment:
Also, consider using match groups instead of string.split down the line

Reply:
Considered it, doesn't play nice with Typescript. Match doesn't work with capture groups when using the global flag. I would like to use `matchAll, but `String.prototype.matchAll` is relatively new, and Typescript doesn't recognize that it exists. I tried for about 20 minutes to adjust the TS config so it would work, but none of the fixes I found online worked, so I just gave up. At least the fact that it matches the regex ensures that by the time it reaches the string.split, it's in the expected format.

Reply:
Add a TODO then :)